whois-api/src/test/java/net/ripe/db/whois/api/rest/OverrideWithNonTrustedIpTestIntegration.java

```diff
@@ -0,0 +1,211 @@
+            assertThat(whoisResources.getTermsAndConditions().getHref(), is(WhoisResources.TERMS_AND_CONDITIONS));
+        }
+    }
+
+    @Test
+    public void create_person_with_override_with_sso_wrong_user_fails() {
+        try {
+            RestTest.target(getPort(), "whois/test/person")
+                    .queryParam("override", "person,zoh,reason")
+                    .request(MediaType.APPLICATION_XML)
+                    .cookie("crowd.token_key", "db_e2e_1")
+                    .post(Entity.entity(map(PAULETH_PALTHEN), MediaType.APPLICATION_XML), WhoisResources.class);
+
+            fail();
+        } catch (NotAuthorizedException ex) {
+            final WhoisResources whoisResources = ex.getResponse().readEntity(WhoisResources.class);
+            RestTest.assertErrorCount(whoisResources, 1);
+            RestTest.assertErrorMessage(whoisResources, 0, "Error", "Override only allowed by database administrators");
+            assertThat(whoisResources.getTermsAndConditions().getHref(), is(WhoisResources.TERMS_AND_CONDITIONS));
+        }
+    }
+
+    @Test
+    public void create_person_with_override_with_sso_wrong_domain_fails() {
+        try {
+            RestTest.target(getPort(), "whois/test/person")
+                    .queryParam("override", "person,zoh,reason")
+                    .request(MediaType.APPLICATION_XML)
+                    .cookie("crowd.token_key", "valid-token")
+                    .post(Entity.entity(map(PAULETH_PALTHEN), MediaType.APPLICATION_XML), WhoisResources.class);
+
+            fail();
+        } catch (NotAuthorizedException ex) {
+            final WhoisResources whoisResources = ex.getResponse().readEntity(WhoisResources.class);
+            RestTest.assertErrorCount(whoisResources, 1);
+            RestTest.assertErrorMessage(whoisResources, 0, "Error", "Override only allowed by database administrators");
+            assertThat(whoisResources.getTermsAndConditions().getHref(), is(WhoisResources.TERMS_AND_CONDITIONS));
+        }
+    }
+
+    @Test
+    public void create_person_with_override_with_sso_succeeds() {
+        final WhoisResources whoisResources = RestTest.target(getPort(), "whois/test/person")
+                .queryParam("override", encode("db_e2e_1,zoh,reason {notify=false}"))
+                .request()
+                .cookie("crowd.token_key", "db_e2e_1")
+                .post(Entity.entity(map(PAULETH_PALTHEN), MediaType.APPLICATION_XML), WhoisResources.class);
+
+        final WhoisObject object = whoisResources.getWhoisObjects().get(0);
+        assertThat(whoisResources.getErrorMessages().get(0).getText(), is("Authorisation override used"));
+        assertThat(object.getType(), is("person"));
+        assertThat(object.getPrimaryKey(), contains(new Attribute("nic-hdl", "PP1-TEST")));
+    }
+
+    @Test
+    public void delete_person_with_override_with_sso_succeeds() {
```

Comment:
This tests the same as create (ip not trusted, valid sso + override) so this test is a bit redundant?

Reply:
Agree, I was just including create, update and delete. If there is no need I can remove these two.